vllm/entrypoints/openai/api_server.py

```diff
@@ -625,6 +625,26 @@ async def reset_prefix_cache(raw_request: Request):
         await engine_client(raw_request).reset_prefix_cache()
         return Response(status_code=200)
 
+    @router.post("/sleep")
+    async def sleep(raw_request: Request):
+        # get POST params
+        level = raw_request.query_params.get("level", "1")
+        logger.info("sleep the engine with level %s", level)
+        await engine_client(raw_request).sleep(int(level))
+        # when we return a response, the sleep command
+        # is sent but does not finish yet.
+        # TODO: we should wait for the sleep to finish and then return
```

Comment:
Not sure if it's worth updating this comment since it does now wait in V1?

Reply:
updated in [e8416d9](https://github.com/vllm-project/vllm/pull/12987/commits/e8416d9d4a7be65ffdab658bf0d5dccd69e03070) now.